### PR TITLE
Land enableNativeEventPriorityInference

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

After a successful experiment, this diff lands a change that updates native events that fire outside of the React event system so that they use the event type to schedule updates at the correct priority, which would match the priority and behvior used by React if the event was fired in our event system.

See https://github.com/facebook/react/pull/20748 for more info.

### CODE REVIEW

**Review by acdlite — APPROVED** [review]

**Review by gaearon — APPROVED** [review]

**Comment by rickhanlonii** [comment]
Waiting for an internal diff to land before landing this.

**Pull request merged**
